File: tests/core/resources/diff/npm/function/left.js
function fibonacci(n) {
    if (n <= 0) return 0;
    if (n === 1) return 1;
  
    let prev = 0, curr = 1;
  
    for (let i = 2; i <= n; i++) {
      [prev, curr] = [curr, prev + curr];
    }
  
    return curr;
  }
  
function main() {
    const args = process.argv.slice(2);
    if (args.length !== 1) {
      console.error('Usage: fib n');
      process.exit(1);
    }
  
    const n = parseInt(args[0], 10);
    if (isNaN(n) || n < 0) {
      console.error('Error: a non-negative index is required');
      process.exit(1);
    }

    const result = fibonacci(n)
    console.log(`${result}`);
  }
  
  main();

File: tests/core/resources/diff/go/function/left.go
package main

import (
	"fmt"
	"os"
	"strconv"
)

func fibonacci(n int) int {
	if n <= 0 {
		return 0
	}
	if n == 1 {
		return 1
	}

	prev, curr := 0, 1
	for i := 2; i <= n; i++ {
		prev, curr = curr, prev+curr
	}

	return curr
}

func main() {
	if len(os.Args) != 2 {
		fmt.Println("Usage: fib n")
		os.Exit(1)
	}

	n, err := strconv.Atoi(os.Args[1])
	if err != nil || n < 0 {
		fmt.Println("Error: a non-negative integer is required.")
		os.Exit(1)
	}

	result := fibonacci(n)
	fmt.Printf("%d\n", result)
}
